### DISCUSSION TITLE
Improving performance of developer console

### DISCUSSION THREAD

**Discussion by doonv** [emilk/egui#5404] [Q&A]
I'm working on a developer console for Bevy using `bevy_egui`. And I'm experiencing an issue where the FPS goes down when there are more logs, and this is probably because I am rendering **all** the log messages at once, but because all the logs are rendered within a `ScrollArea`, only a small portion of the logs are actually shown, but all of them are still rendered, which is probably causing the lag.

So my question is, how do I skip rendering the logs that aren't visible within the `ScrollArea`? (While also not shrinking the total scroll area)

Link to the code that renders the logs and `ScrollArea`: https://github.com/doonv/bevy_dev_console/blob/98355f16b7d83ad0db983f2204426079780aa2ae/src/ui.rs#L184

**Comment by voidburn** [comment]
You can use the `show_rows` method to render only the visible entries (assuming they're all the same height): https://docs.rs/egui/latest/egui/containers/scroll_area/struct.ScrollArea.html#method.show_rows

**Reply by doonv** [reply]
> assuming they're all the same height

I forgot to mention that they are not all the same height (because of line wrapping). Is there something I can do about that? (except for disabling line wrapping)

**Reply by voidburn** [reply]
Haven't yet figured that out!
